pphysx3d/src/shapes/cube.rs

```diff
@@ -1,137 +0,0 @@
-    fn as_cube(&self) -> Result<&Cube, ()> {
-        Ok(&self)
-    }
-
-    fn as_plane(&self) -> Result<&super::plane::Plane, ()> {
-        Err(())
-    }
-}
-
-impl RayCast for Cube {
-    fn ray_cast(&self, pos: &Isometry3<f32>, ray: &Ray) -> RayCastResult {
-        // Based on https://learning.oreilly.com/library/view/game-physics-cookbook/9781787123663/ch10s04.html
-
-        let mut result = RayCastResult::new();
-
-        let size = self.half_extents;
-
-        let x_rotated: UnitVector3<f32> = UnitVector3::new_normalize(pos.rotation * Vector3::x());
-        let y_rotated: UnitVector3<f32> = UnitVector3::new_normalize(pos.rotation * Vector3::y());
-        let z_rotated: UnitVector3<f32> = UnitVector3::new_normalize(pos.rotation * Vector3::z());
-
-        let distance_to_center: Vector3<f32> = Point3::from(pos.translation.vector) - ray.origin();
-
-        let mut direction_proj = [
-            x_rotated.dot(ray.direction()),
-            y_rotated.dot(ray.direction()),
-            z_rotated.dot(ray.direction()),
-        ];
-
-        let distance_proj = [
-            x_rotated.dot(&distance_to_center),
-            y_rotated.dot(&distance_to_center),
-            z_rotated.dot(&distance_to_center),
-        ];
-
-        // stores the min/max intersection points for all three axes
-        let mut t = [0f32; 6];
-
-        for i in 0..3 {
-            // if ray direction is parallel to axis
-            if direction_proj[i] == 0.0 {
-                // if ray origin is not inside axis-slab
-                if -distance_proj[i] - size[i] > 0.0 || -distance_proj[i] + size[i] < 0.0 {
-                    return result; // ray must hit all three slabs
-                }
-                direction_proj[i] = 0.00001 // Almost zero to avoid div-by-zero
-            }
-            t[i * 2 + 0] = (distance_proj[i] + size[i]) / direction_proj[i]; // min
-            t[i * 2 + 1] = (distance_proj[i] - size[i]) / direction_proj[i]; // max
-        }
-        // entry and exit points
-        let tmin = t[0].min(t[1]).max(t[2].min(t[3])).max(t[4].min(t[4]));
-        let tmax = t[0].max(t[1]).min(t[2].max(t[3])).min(t[4].max(t[4]));
-
-        // if cube is "behind" ray
-        if tmax < 0.0 {
-            return result;
-        }
-        // if not intersecting
-        if tmin > tmax {
-            return result;
-        }
-
-        result.distance = tmin;
-
-        // if ray starts inside cube
-        if tmin < 0.0 {
-            result.distance = tmax
-        }
-
-        result.contact_point = ray.origin() + ray.direction().scale(result.distance);
-
-        // match which normal was hit
-        let normals: [UnitVector3<f32>; 6] = [
-            x_rotated, -x_rotated, y_rotated, -y_rotated, z_rotated, -z_rotated,
-        ];
-        for i in 0..6 {
-            if t[i] == result.distance {
-                result.normal = normals[i];
-            }
-        }
-
-        result.hit = true;
-
-        return result;
-    }
-}
```

Comment:
Är det smart att ta.bort filerna och koden som berör kuben? Vad ska man göra om man vill fortsätta bygga på det?

Reply:
Det har vi dev branchen för är tanken

Reply:
Har vi en dev branch? 👀

Reply:
yeye flyttade allt dit för att göra repot mer clean

Reply:
Damn lets yeet this one then